### ISSUE TITLE
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url

### ISSUE THREAD

**Issue by kerryshireman** [cyberjunky/python-garminconnect#235]
Hi all, I just started receiving an error today using this lib. I was on the 0.2.14 release prior to today, so I updated to the latest 0.2.21 release. I see that MFA is now included in the example code, so I incorporated the latest init_api function.  Username & password challenge seems to work with the login method, as I am challenged for an MFA one-time code. After entering the code, I then get an exception: 
```
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://sso.garmin.com/sso/signin?id=gauth-widget&embedWidget=true&gauthHost=https://sso.garmin.com/sso/embed&service=https://sso.garmin.com/sso/embed&source=https://sso.garmin.com/sso/embed&redirectAfterAccountLoginUrl=https://sso.garmin.com/sso/embed&redirectAfterAccountCreationUrl=https://sso.garmin.com/sso/embed
```

I didn't see any related active issues. I've made no other changes to the Python setup using garminconnect for about 10 months. I've confirmed the same credentials work fine on the Garmin Connect website. Is anyone else having the same issue of late?

**Comment by kerryshireman** [comment]
I dropped down to simply using the example.py file. Each time I login, the code setting the [display_name](https://github.com/cyberjunky/python-garminconnect/blob/master/garminconnect/__init__.py#L230) triggers this error

```
  File "/Users/{path-to-venv}/lib/python3.11/site-packages/garminconnect/__init__.py", line 230, in login
    self.display_name = self.garth.profile["displayName"]
                        ^^^^^^^^^^^^^^^^^^
  File "/Users/{path-to-venv}/lib/python3.11/site-packages/garth/http.py", line 104, in profile
    assert isinstance(self._profile, dict)
AssertionError
```

**Comment by pblocz** [comment]
I am having the same error for the AssertionError, seems someone has reported it as an API change by Garmin in the garth library https://github.com/matin/garth/issues/73

**Comment by kerryshireman** [comment]
Good find, @pblocz, it's so fun when Garmin changes things. I'll leave this issue open for the moment, but assuming a garth fix is forthcoming, the [pyproject.toml here](https://github.com/cyberjunky/python-garminconnect/blob/master/pyproject.toml#L9) should pick it up.

**Comment by RobertWojtowicz** [comment]
> I dropped down to simply using the example.py file. Each time I login, the code setting the [display_name](https://github.com/cyberjunky/python-garminconnect/blob/master/garminconnect/__init__.py#L230) triggers this error
> 
> ```
>   File "/Users/{path-to-venv}/lib/python3.11/site-packages/garminconnect/__init__.py", line 230, in login
>     self.display_name = self.garth.profile["displayName"]
>                         ^^^^^^^^^^^^^^^^^^
>   File "/Users/{path-to-venv}/lib/python3.11/site-packages/garth/http.py", line 104, in profile
>     assert isinstance(self._profile, dict)
> AssertionError
> ```

Unfortunately I have the same

BR,
Robert

**Comment by yorkshirelad99** [comment]
confirm the same issue, tried on 2 machines and having the same problem

**Comment by enydog** [comment]
The issue seems to be with garth, as a simple example triggers the same error:
return self.profile["userName"]

test:
import garth
from getpass import getpass

email = input("Enter email address: ")
password = getpass("Enter password: ")
# If there's MFA, you'll be prompted during the login
garth.login(email, password)
sleep = garth.connectapi(
    f"/wellness-service/wellness/dailySleepData/{garth.client.username}",
    params={"date": "2023-07-05", "nonSleepBufferMinutes": 60},
)
list(sleep.keys())

**Comment by psdupvi** [comment]
This issue can be fixed (thanks to the good work of the people over at [garth](https://github.com/matin/garth/issues/73)) by going into the Garth package file you have as part of your environment and changing the following line in `garth/http.py`

Line 15-17

From:
```python
USER_AGENT = {
    "User-Agent": (
        "Mozilla/5.0 (iPhone; CPU iPhone OS 16_5 like Mac OS X) "
        "AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148"
    ),
}
```

to 

```python
USER_AGENT = {
    "User-Agent": ("GCM-iOS-5.7.2.1"),
}
```

There's a PR on that repo to update the user agent from our great maintainer @cyberjunky , but while we wait for that to get merged and everything to update, this will work

**Comment by kerryshireman** [comment]
It appears a fix is imminent.

@cyberjunky already merged a USER_AGENT fix in their fork:  https://github.com/cyberjunky/garth/pull/1
My local tests updating the user agent restored API functionality.
Said fix is being merged into garth proper: https://github.com/matin/garth/pull/74

Assuming a garth release is forthcoming, reinstalling python-garminconnect deps should pull in the fix.

 Thanks all for such a fast response! No one likes it when their fitness automations are brokwn.

**Comment by LilDinker** [comment]
Thanks guys!

**Comment by enydog** [comment]
thanks.

**Comment by pblocz** [comment]
I just tried it, but while https://github.com/matin/garth/pull/74 is being merged and a new release of garth is created, this patch should do it:

```python
import garth.http
garth.http.USER_AGENT = {"User-Agent": "GCM-iOS-5.7.2.1",}
```

**Comment by cyberjunky** [comment]
Temp fixed in version 0.2.22!
